docs/kcl_mod.md

```diff
@@ -0,0 +1,70 @@
+# kcl.mod
+
+## Background
+kpm needs a configuration file for KCL project, which is responsible for marking the root directory, determining dependency versions, configuring compilation options or others.
+
+## Capacity
+
+- name: the name of the current project.
+- version: the version of the current project.
+- dependencies: the dependencies of the current project.
+
+## How to write kcl.mod
+The contents of the kcl.mod file are formatted using the syntax of [toml](https://toml.io/en/).
+
+### name, version and compiler version.
+```toml
+[package]
+name = "kcl"
+version = "0.0.0"
+edition = "0.0.0"
+```
+
+### dependencies
+
+- Registry
+
+```toml
+[dependencies]
+xxx = "0.0.1"
+```
+
+- Local path
+
+```toml
+# Find the kcl.mod under "./xxx/xxx".
+[dependencies]
+xxxx = {path = "./xxx/xxx",version = "0.4.5"} 
+```
+
+- Git 
+```toml
+[dependencies]
+xxxx = {git = "<github url>", version = "0.4.5"} 
+xxxx = {git = "<github url>", commit = "h87h8f78"} 
+xxxx = {git = "<github url>", branch = "main"} 
+```
+
+## Workspaces
+```toml
+[workspace]
+members = [
+    # subdirectory relative to the current directory, 
+    # which must contains kcl.mod.
+    "xxx" 
+]
+```
+
+## kpm.lock
```

Comment:
+ Just use `kcl.mod.lock`
+ Missing specification definition of lock file.

Reply:
fixed.